torch/distributed/pipelining/schedules.py

```diff
@@ -955,3 +956,151 @@ def backward_stage_index(step):
         for _ in range(self.pp_group_size - rank - 1):
             rank_ops.append(None)
         return rank_ops
+
+
+class ScheduleFlexibleInterleaved1F1B(PipelineScheduleMulti):
+    """
+    The Flexible Interleaved 1F1B schedule.
```

Comment:
nit: you almost say this, but maybe be a little more explicit
"""
The Flexible Interleaved 1F1B schedule.

This schedule is mostly similar to the interleaved 1F1B schedule.  It differs by being relaxing the requirement of num_microbatch % pp_size == 0.  

...
"""

Also, is there any tradeoff here?  like if you have num_microbatches == pp_size * 2, should you pick regular interleaved 1f1b? or should you just always use this one?

Reply:
Under the current implementation, they are the same when num_microbatches == pp_size * 2. My thinking of keeping these separate is that we might evolve this to DoraPP and others as a next step so imho would be great to have a baseline 1f1b version instead of consolidating the two from the very beginning. That being said, I'll merge most of the code of the two schedules when I update the PR

Reply:
Yea, seems reasonable to me. If we are going to write several schedules with a lot of similarity then minimizing copy paste via refactoring seems beneficial